datalake/src/main/java/com/sequenceiq/datalake/service/sdx/status/SdxStatusService.java

```diff
@@ -53,11 +57,30 @@
     @Inject
     private Clock clock;
 
+    @Inject
+    private FlowLogService flowLogService;
+
+    @Inject
+    private FlowChainLogService flowChainLogService;
+
     public void setStatusForDatalakeAndNotify(DatalakeStatusEnum status, String statusReason, SdxCluster sdxCluster) {
         setStatusForDatalake(status, statusReason, sdxCluster);
         sdxNotificationService.send(status.getDefaultResourceEvent(), sdxCluster);
     }
 
+    public SdxCluster setStatusForDatalakeAndNotify(DatalakeStatusEnum status, String statusReason, Long datalakeId, String flowId) {
+        FlowLog flowLog = flowLogService.getLastFlowLog(flowId).get();
```

Comment:
`Optional.get()` will fail if the flow doesn't exist. It might worth considering to make this flow based status modification a best effort staff and if a flow log is for some reason not present for the flowId then just use the original status. Wouldn't be good to have an ENGESC for failed resize just because a status update failed.